app/Cmd.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Cmd (
+  CmdOpts (CmdOpts),
+  MongoOpts (MongoOpts)
+  ) where
+
+import           Data.Bifunctor
+import           Data.Maybe
+import           Data.Text        hiding (group)
+import           Database.MongoDB hiding (Group, group)
+import           Options
+
+-------------------------------------------------------------------------------------------------------------------------
+-- Global options
+-------------------------------------------------------------------------------------------------------------------------
+data CmdOpts = CmdOpts {
+  mongoOpts :: MongoOpts,
+  port      :: Int
+  } deriving (Eq, Show)
+
+instance Options CmdOpts where
+  defineOptions = pure CmdOpts
+                  <*> defineOptions
+                  <*> simpleOption "port" 3000 "Port on which to listen for incoming HTTP calls."
+
+-------------------------------------------------------------------------------------------------------------------------
+-- MongoDB options
+-------------------------------------------------------------------------------------------------------------------------
+
+data MongoOpts = MongoOpts {
+  mongoHost :: Host,
+  mongoDb   :: Text,
+  mongoCol  :: Text
```

Comment:
I would prefer to see `newtype`s for these `Text` fields. It's real easy to pass the wrong thing, but a `newtype` will guarantee that your program doesn't confuse ` mongoDb`  or `mongoCol`.

Reply:
Understood. Does that apply to the `Mongo` type in `Main` as well? This one is strictly internal and meant as a convenience - to avoid passing multiple values around, and forcing an `unwrap` call seems counterproductive.

Reply:
`newtype` hygiene, to me, is about protecting the program as it grows. Small programs can get by without any type safety at all. As a program grows and evolves, though, it's easier and easier to run into bugs. I generally try to newtype `Text` values whenever I have some more specific use than "a blob of unstructured human readable text". 

I'd also not use `unwrap` as a generic way to unrwap the newtype, and instead use a named function or a pattern-match as close as possible to the edge of the system.